cpp/src/cider/function/CMakeLists.txt

```diff
@@ -99,6 +99,10 @@ add_custom_target(
   DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/RuntimeFunctions.bc
           ${CMAKE_CURRENT_BINARY_DIR}/ExtensionFunctions.ast)
 
+add_library(cider_runtime_function SHARED ${runtime_function_source_files})
+add_dependencies(cider_runtime_function QueryEngineFunctionsTargets)
+target_link_libraries(cider_runtime_function re2 substrait)
+
 add_library(cider_function SHARED ${function_source_files})
 add_dependencies(cider_function QueryEngineFunctionsTargets)
-target_link_libraries(cider_function nextgen re2 cider_type_plan substrait)
+target_link_libraries(cider_function substrait nextgen)
```

Comment:
what's the difference between cider_runtime_function and cider_function?
why does cider_function depend on nextgen?

Reply:
I moved runtime functions out of it as an independent lib.
Because some SubstritToEu components in cider_function needs ```Expression``` classes that have been merged into nextgen. nextgen and SubstritToEu should be combined further.